src/context_free.ml

```diff
@@ -197,6 +197,48 @@ module Generated_code_hook = struct
     | _ -> t.f context { loc with loc_start = loc.loc_end } x
 end
 
+(* Used to insert error extensions *)
+let wrap_extension : type a. loc:Location.t -> a EC.t -> a -> extension -> a =
+ fun ~loc t original_node extension ->
+  (* Prefixing constructors with the module path is necessary for OCaml < 4.07,
+     see https://github.com/ocaml/ocaml/issues/6852 *)
+  match t with
+  | EC.Class_expr -> Ast_builder.Default.pcl_extension ~loc extension
+  | EC.Class_field -> Ast_builder.Default.pcf_extension ~loc extension
+  | EC.Class_type -> Ast_builder.Default.pcty_extension ~loc extension
+  | EC.Class_type_field -> Ast_builder.Default.pctf_extension ~loc extension
+  | EC.Core_type -> Ast_builder.Default.ptyp_extension ~loc extension
+  | EC.Expression -> Ast_builder.Default.pexp_extension ~loc extension
+  | EC.Module_expr -> Ast_builder.Default.pmod_extension ~loc extension
+  | EC.Module_type -> Ast_builder.Default.pmty_extension ~loc extension
+  | EC.Pattern -> Ast_builder.Default.ppat_extension ~loc extension
+  | EC.Signature_item -> Ast_builder.Default.psig_extension ~loc extension []
+  | EC.Structure_item -> Ast_builder.Default.pstr_extension ~loc extension []
+  | EC.Ppx_import ->
+      (* Insert the error in the type decl manifest *)
```

Comment:
We've had the [same idea](https://github.com/ocaml-ppx/ppxlib/pull/294/files#diff-1039f50ffb020bed079f732077cd7111c5abbf9845a47564d28af250f0e95b98R133) :slightly_smiling_face: 

This function could be in `extension.ml`, in my opinion, but I'm not even sure and that might be bikeshedding: feel free to ignore the suggestion...

Reply:
I moved it in context_free because putting it in extension meant to make it public and I'm not sure we want to.

It is also only used for errors and is therefore very specific.